### PR TITLE
Add unit tests for the sset package

### PR DESCRIPTION
**Pull request by sebgl**
A bunch of functions were left with no unit test. Let's fix that!

Relates https://github.com/elastic/cloud-on-k8s/issues/1287

### CODE REVIEW

File: operators/pkg/controller/elasticsearch/sset/getter.go [hunk 1]
```diff
@@ -15,3 +17,20 @@ func Replicas(statefulSet appsv1.StatefulSet) int32 {
...
+// UpdatePartition returns the updateStrategy.Partition index, or falls back to the number of replicas if not set.
+func UpdatePartition(statefulSet appsv1.StatefulSet) int32 {
+	rollingUpdate := statefulSet.Spec.UpdateStrategy.RollingUpdate
+	if rollingUpdate != nil && rollingUpdate.Partition != nil {
+		return *rollingUpdate.Partition
+	}
+	if statefulSet.Spec.Replicas != nil {
+		return *statefulSet.Spec.Replicas
+	}
+	return 0
```
Comment (david-kow) [inline operators/pkg/controller/elasticsearch/sset/getter.go]: Should this be -1? I assume partitions are numbered from 0 so right now there is no result difference between error path and happy path - could that be an issue?

Comment (sebgl) [reply operators/pkg/controller/elasticsearch/sset/getter.go]: I think this is a rather sane default. We always explicitly set replicas and update partition, but if a user happens to set it back to nil explicitly (not sure why they would do that), we consider all pods can be upgraded.

Returning -1 here could have other unintended consequences I think (not sure what happens if we explicitly update the partition to -1 for example).

Comment (david-kow) [reply operators/pkg/controller/elasticsearch/sset/getter.go]: -1 would skip entire loop, I'm not sure where 0 with error would end up. If you feel it's safe and clear then I'm fine with 0 though:)

File: operators/pkg/controller/elasticsearch/sset/getter.go [hunk 2]
```diff
@@ -15,3 +17,20 @@ func Replicas(statefulSet appsv1.StatefulSet) int32 {
 	}
 	return 0
 }
+
+// UpdatePartition returns the updateStrategy.Partition index, or falls back to the number of replicas if not set.
+func UpdatePartition(statefulSet appsv1.StatefulSet) int32 {
```
Comment (david-kow) [inline operators/pkg/controller/elasticsearch/sset/getter.go]: Why rename?

Comment (sebgl) [reply operators/pkg/controller/elasticsearch/sset/getter.go]: To better match the other function in this file: `Replicas()`. So we have:

```go
replicas := sset.Replicas(s)
updatePartition := sset.UpdatePartition(s)
```

Do you think `GetReplicas` and `GetUpdatePartition` would be better?

Comment (david-kow) [reply operators/pkg/controller/elasticsearch/sset/getter.go]: My only complain is that I read update as a verb and it feels surprising it doesn't update anything :) Partition? But this reads as a verb too... CurrentPartition?

Comment (sebgl) [reply operators/pkg/controller/elasticsearch/sset/getter.go]: Agreed, this is confusing.
Maybe using `Get` for both is better.

**Pull request merged**